os/fs/smartfs/smartfs_utils.c

```diff
@@ -1211,7 +1230,7 @@ int smartfs_find_availableentry(struct smartfs_mountpt_s *fs, uint16_t *parentdi
 				direntry->dsector = psector;
 				direntry->doffset = offset;
 				direntry->flags = entry->flags;
-				*parentdirsector = direntry->dsector;
+				direntry->last_dsector = direntry->dsector;
```

Comment:
we don't have to update dsector & last_dsector here, it must be out side of while.
assume that sector is chained #3 -> #12  and they are filled with entries(total 24 if sector size is 512)
if sector 13 is empty, anyway it will find from sector 3
At the start of loop, both of dsector & last_dsector are probably 3.
After first loop,  dsector become 12, but last_dsector is still 3..
and second loop, it will break loop. dsector become maybe 0xffff
and dsector changed to new alloc sector by smartfs_createdirentry, it may alloc sector 13.
So it will chain sector 3 to 13 but sector 3 has next sector(12) already,
so some of bits will be corrupted at that time.

Reply:
> dsector become maybe 0xffff

This cannot happen because
`direntry->dsector = psector;
		psector = nextsector;`
So, psector can become 0xffff, but dsector can not

So at the end of checking parent chain, dsector = 12, psector = 0xffff, last_dsector = 3

Now, Before going for createentry
`direntry->last_dsector = direntry->dsector;`
So, last_dsector = 12 now
Dsector allocated 13 by createentry
13 will be chained to 12 during sync.
And not 
> So it will chain sector 3 to 13